webapps-backup/src/test/java/io/camunda/webapps/backup/repository/opensearch/OpensearchBackupRepositoryTest.java

```diff
@@ -0,0 +1,428 @@
+  }
+
+  @Test
+  void shouldReturnBackupStateFailedWhenSnapshotIsPartialCompleted() throws IOException {
+    final var firstSnapshotInfo =
+        SnapshotInfo.of(
+            bi ->
+                defaultFields(bi)
+                    .uuid("uuid")
+                    .state(SnapshotState.PARTIAL.name())
+                    .startTimeInMillis(Long.toString(Instant.now().toEpochMilli())));
+    when(openSearchClient.snapshot().get((GetSnapshotRequest) any()))
+        .thenReturn(
+            GetSnapshotResponse.of(b -> defaultFields(b).snapshots(List.of(firstSnapshotInfo))));
+    // Test
+    final var backupState = repository.getBackupState("repository-name", 5L);
+    Assertions.assertThat(backupState.getState()).isEqualTo(BackupStateDto.FAILED);
+  }
+
+  @Test
+  void shouldReturnBackupStateFailedCompleted() throws IOException {
+    final var firstSnapshotInfo =
+        SnapshotInfo.of(
+            bi ->
+                defaultFields(bi, new Metadata(5L, "1", 1, 2))
+                    .uuid("uuid")
+                    .state(SnapshotState.SUCCESS.name())
+                    .startTimeInMillis(Long.toString(Instant.now().toEpochMilli())));
+    final var lastSnapshotInfo =
+        SnapshotInfo.of(
+            bi ->
+                defaultFields(bi, new Metadata(5L, "1", 2, 2))
+                    .uuid("uuid")
+                    .state(SnapshotState.SUCCESS.name())
+                    .startTimeInMillis(Long.toString(Instant.now().toEpochMilli())));
+    when(openSearchClient.snapshot().get((GetSnapshotRequest) any()))
+        .thenReturn(
+            GetSnapshotResponse.of(
+                b -> defaultFields(b).snapshots(List.of(firstSnapshotInfo, lastSnapshotInfo))));
+    // Test
+    final var backupState = repository.getBackupState("repository-name", 5L);
+    Assertions.assertThat(backupState.getState()).isEqualTo(BackupStateDto.COMPLETED);
+  }
+
+  @Test
+  void shouldReturnBackupStateFailedWhenSnapshotIsFailed() throws IOException {
+    final var firstSnapshotInfo =
+        SnapshotInfo.of(
+            bi ->
+                defaultFields(bi)
+                    .uuid("uuid")
+                    .state(SnapshotState.FAILED.name())
+                    .startTimeInMillis(Long.toString(Instant.now().toEpochMilli())));
+    when(openSearchClient.snapshot().get((GetSnapshotRequest) any()))
+        .thenReturn(
+            GetSnapshotResponse.of(b -> defaultFields(b).snapshots(List.of(firstSnapshotInfo))));
+    // Test
+    final var backupState = repository.getBackupState("repository-name", 5L);
+    Assertions.assertThat(backupState.getState()).isEqualTo(BackupStateDto.FAILED);
+  }
+
+  private SnapshotInfo.Builder defaultFields(final SnapshotInfo.Builder b) {
+    return defaultFields(b, new Metadata(1L, "1", 1, 1));
+  }
+
+  private SnapshotInfo.Builder defaultFields(
+      final SnapshotInfo.Builder b, final Metadata metadata) {
+    return b.dataStreams(List.of())
+        .indices(List.of())
```

Comment:
these fields must be present in the builder, as validation checks for their presence when building the object